### PR TITLE
Add FXIOS-7188 [v121] Add & update search settings strings

### PR DESCRIPTION
**Pull request by adudenamedruby**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7188)

## :bulb: Description
This was missed, but it's ok, because we pushed string export to Monday. I will do better next time - please don't take away my ice cream.

PR: Update the organization of the search strings. Added tablenames. Updated use locations. Added new strings for privacy.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by adudenamedruby** [comment]
Screenshot 2023-11-15 at 2 06 44 PM

lololol

File: Client/Frontend/Strings.swift [hunk 1]
```diff
@@ -1624,6 +1624,47 @@ extension String {
                     comment: "Title for the page where the user sign in to their Firefox Sync account.")
             }
         }
+
+        public struct Search {
+            public static let Title = MZLocalizedString(
```
Comment (lmarceau) [inline Client/Frontend/Strings.swift:1629]: Are those strings really needed for 121? String freeze for that version was yesterday.

File: Client/Frontend/Strings.swift [hunk 2]
```diff
@@ -1624,6 +1624,47 @@ extension String {
...
+                key: "Settings.Search.PageTitle.v121",
+                tableName: "Settings",
+                value: "Search",
+                comment: "Navigation title for search page in the Settings menu.")
+            public static let ShowSearchSuggestions = MZLocalizedString(
+                key: "Settings.Search.ShowSuggestions.v121",
+                tableName: "Settings",
+                value: "Show Search Suggestions",
+                comment: "Label for show search suggestions setting, in the Search Settings page.")
```
Comment (lmarceau) [inline Client/Frontend/Strings.swift:1638]: Reads better that way?
```suggestion
                comment: "Label for the `show search suggestions` setting, in the Search Settings page.")
```

File: Client/Frontend/Strings.swift [hunk 3]
```diff
@@ -1624,6 +1624,47 @@ extension String {
...
+            public static let DefaultSearchEngineTitle = MZLocalizedString(
+                key: "Settings.Search.DefaultSearchEngine.Title.v121",
+                tableName: "Settings",
+                value: "Default Search Engine",
+                comment: "Title for default search engine settings section in the Search page in the Settings menu.")
```
Comment (lmarceau) [inline Client/Frontend/Strings.swift:1643]: Small suggestion again, it just feels like it reads better that way if you don't know what this is about. I won't make the comments on the other strings, but I would make the same comment 🤷‍♀️
```suggestion
                comment: "Title for the `default search engine` settings section in the Search page in the Settings menu.")
```

File: Client/Frontend/Strings.swift [hunk 4]
```diff
@@ -1624,6 +1624,47 @@ extension String {
...
+            public static let QuickSearchEnginesTitle = MZLocalizedString(
+                key: "Settings.Search.QuickEnginesTitle.v121",
+                tableName: "Settings",
+                value: "Quick-Search Engines",
+                comment: "Title for quick-search engines settings section in the Search page in the Settings menu.")
+            public static let PrivateSessionTitle = MZLocalizedString(
+                key: "Settings.Search.PrivateSession.Title.v121",
+                tableName: "Settings",
+                value: "Private Session",
+                comment: "Title for the Private Session settings section in the Search page in the Settings menu.")
```
Comment (lmarceau) [inline Client/Frontend/Strings.swift:1653]: Ah, see you had `the` already here :smile:
```suggestion
                comment: "Title for the `Private Session` settings section in the Search page in the Settings menu.")
```

File: Client/Frontend/Strings.swift [hunk 5]
```diff
@@ -1624,6 +1624,47 @@ extension String {
...
+            public static let PrivateSessionSetting = MZLocalizedString(
+                key: "Settings.Search.PrivateSession.Setting.v121",
+                tableName: "Settings",
+                value: "Turn off suggestions in private browsing",
+                comment: "Label for toggle. Explains that, in private browsing mode, search suggestions, the suggestions appearing at the top of the search bar, can be toggled on or off. Located in the Private Session section in the Search page in the Settings menu.")
```
Comment (lmarceau) [inline Client/Frontend/Strings.swift:1658]: Lots of commas?
```suggestion
                comment: "Label for toggle. Explains that in private browsing mode, the search suggestions which appears at the top of the search bar, can be toggled on or off. Located in the Private Session section in the Search page in the Settings menu.")
```

File: Client/Frontend/Strings.swift [hunk 6]
```diff
@@ -5224,35 +5265,6 @@ extension String {
         comment: "Label for Cancel button")
 }
 
-// MARK: - SearchSettings
-extension String {
-    public static let SearchSettingsTitle = MZLocalizedString(
```
Comment (lmarceau) [inline Client/Frontend/Strings.swift:5229]: We can't remove strings that are still used in previous version, until that string is entirely released in the last version it was used in. This is an ask by the l10n team. Please see `How to remove a string` in [this wiki](https://github.com/mozilla-mobile/firefox-ios/wiki/How-to-add-and-modify-Strings#how-to-remove-a-string).

File: Client/Frontend/Strings.swift (hunk 1 above)
Comment (adudenamedruby) [reply Client/Frontend/Strings.swift:1629]: String freeze for that version was pushed to Monday due to some confusion with stuff. So we can still get these in. Yes. :D

**Review by lmarceau — APPROVED** [review]

**Pull request merged**